src/Microsoft.Health.TaskManagement.UnitTests/TestQueueClient.cs

```diff
@@ -226,5 +226,10 @@ public Task<bool> PutJobHeartbeatAsync(JobInfo jobInfo, CancellationToken cancel
 
             return Task.FromResult(cancel);
         }
+
+        public Task<string> PeekAsync(byte queueType, CancellationToken cancellationToken)
+        {
+            return Task.FromResult(jobInfos.Where(j => j.QueueType == queueType && (j.Status == JobStatus.Created || j.Status == JobStatus.Running)).ToList().FirstOrDefault().Id.ToString());
```

Comment:
This throws when the jobInfos.Count == 0. You'll need to revise as:
FirstOrDefault()?.Id.ToString()

Reply:
Thanks, I updated it.